pkg/api/v20230701preview/openshiftcluster.go

```diff
@@ -78,6 +78,7 @@ type ProvisioningState string
 const (
 	ProvisioningStateCreating      ProvisioningState = "Creating"
 	ProvisioningStateUpdating      ProvisioningState = "Updating"
+	ProvisioningStateCancelled     ProvisioningState = "Cancelled"
```

Comment:
What's the criteria for "Cancelled"?  We should add it to our top-level openshiftcluster doc.  We should also add logic or a TODO comment stating we do not use it yet.

Reply:
ARM defines only 3 terminal states. Succeed, Failed, Cancelled. For linting purposes upstream requires at least these three to be present. More info here: https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/async-operations

We can certainly add a TODO comment - but I don't think we ever intend to use this status in our implementation, but it should be included so that upstreaming doesn't nit or have its linting fail. Let me know if you think we should add it still.

Reply:
Pull it into the top-level and throw a todo around it please :). 

Just in case we ever end up using it, our converters won't go nuts.